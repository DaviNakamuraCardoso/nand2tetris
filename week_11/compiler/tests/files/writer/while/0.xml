<keyword>while</keyword>
<symbol>(</symbol>
<integerConstant>2</integerConstant>
<symbol>&gt;</symbol>
<integerConstant>1</integerConstant>
<symbol>)</symbol>
<symbol>{</symbol>
<keyword>do</keyword>
<identifier>SouthAmerica</identifier>
<symbol>.</symbol>
<identifier>takeBack</identifier>
<symbol>(</symbol>
<stringConstant>Samba</stringConstant>
<symbol>)</symbol>
<symbol>;</symbol>
<symbol>}</symbol>
