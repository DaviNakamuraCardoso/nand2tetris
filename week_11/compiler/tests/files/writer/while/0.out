label LABEL_0
push constant 2
push constant 1
gt
not
if-goto LABEL_1
push constant 5
call String.new 1
push constant 83
call String.appendChar 2
push constant 97
call String.appendChar 2
push constant 109
call String.appendChar 2
push constant 98
call String.appendChar 2
push constant 97
call String.appendChar 2
call SouthAmerica.takeBack 1
pop temp 0
goto LABEL_0
label LABEL_1
